pkg/elevator/internal/controllers/tenantobj_controller.go

```diff
@@ -95,80 +100,21 @@ func (r *TenantObjReconciler) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 func (r *TenantObjReconciler) reconcileTenantObj(
-	ctx context.Context, tenantObj *unstructured.Unstructured,
-	config catalogv1alpha1.VersionExposeConfig,
+	ctx context.Context, tenantObj, providerObj *unstructured.Unstructured,
+	statusFields []catalogv1alpha1.FieldPath,
 ) error {
-	desiredProviderObj := r.newProviderObject()
-	desiredProviderObj.SetName(tenantObj.GetName())
-	desiredProviderObj.SetNamespace(tenantObj.GetNamespace())
-
-	err := controllerutil.SetControllerReference(
-		tenantObj, desiredProviderObj, r.Scheme)
-	if err != nil {
-		return fmt.Errorf("set controller reference: %w", err)
-	}
-
-	// prepare config
-	statusFields, otherFields := elevatorutil.SplitStatusFields(config.Fields)
-
-	// Lookup current instance
-	currentProviderObj := r.newProviderObject()
-	err = r.Get(ctx, types.NamespacedName{
-		Name:      desiredProviderObj.GetName(),
-		Namespace: desiredProviderObj.GetNamespace(),
-	}, currentProviderObj)
-	if err != nil && !errors.IsNotFound(err) {
-		return fmt.Errorf("getting %s: %w", r.ProviderGVK.Kind, err)
-	}
-
-	if errors.IsNotFound(err) {
-		// Create the Provider Obj
-		if err = elevatorutil.CopyFields(tenantObj, desiredProviderObj, otherFields); err != nil {
-			return fmt.Errorf("copy fields: %w", err)
-		}
-
-		if err = r.Create(ctx, desiredProviderObj); err != nil {
-			return fmt.Errorf("creating %s: %w", r.ProviderGVK.Kind, err)
-		}
-		return nil
+	if err := r.Patch(ctx, providerObj, client.Apply, elevatorutil.FieldOwner); err != nil {
+		return err
 	}
-
-	// Make sure we take ownership of the provider instance,
-	// if the OwnerReference is not yet set.
-	// Note:
-	// this will raise an error,
-	// if the provider object is already owned by someone else
-	err = controllerutil.SetControllerReference(
-		tenantObj, currentProviderObj, r.Scheme)
-	if err != nil {
-		return fmt.Errorf("set controller reference: %w", err)
-	}
-
-	// Update existing provider instance
-	if err = elevatorutil.CopyFields(tenantObj, currentProviderObj, otherFields); err != nil {
-		return fmt.Errorf(
-			"copy fields from %s to %s: %w",
-			r.TenantGVK.Kind, r.ProviderGVK.Kind, err)
-	}
-	if err = r.Update(ctx, currentProviderObj); err != nil {
-		return fmt.Errorf("updating %s: %w", r.ProviderGVK.Kind, err)
-	}
-
 	// Sync status from provider to tenant instance
-	if err = elevatorutil.CopyFields(currentProviderObj, tenantObj, statusFields); err != nil {
+	if err := elevatorutil.CopyFields(providerObj, tenantObj, statusFields); err != nil {
 		return fmt.Errorf(
 			"copy status fields from %s to %s: %w",
 			r.ProviderGVK.Kind, r.TenantGVK.Kind, err)
 	}
-	if err = util.UpdateObservedGeneration(currentProviderObj, tenantObj); err != nil {
```

Comment:
Why this is removed?

Reply:
It is assuming the tenant CRD has observed generation in the first place. I don't see this being necessary since it can easily be accomplished by defining the appropriate exposed field.